app/src/processing/app/EditorConsole.java

```diff
@@ -110,18 +113,86 @@ public EditorConsole() {
+        if (e.getWheelRotation() < 0) {
+          base.handleFontSizeChange(1);
+        } else {
+          base.handleFontSizeChange(-1);
+        }
+      } else {
+        e.getComponent().getParent().dispatchEvent(e);
+      }
+    });
+
+    // Add "CTRL (SHIFT) =/+" and "CTRL -" hotkeys for font size adjustment.
+    consoleTextPane.addKeyListener(new KeyAdapter() {
+      @Override
+      public void keyPressed(KeyEvent e) {
+        if (e.getModifiersEx() == KeyEvent.CTRL_DOWN_MASK
+            || e.getModifiersEx() == (KeyEvent.CTRL_DOWN_MASK | KeyEvent.SHIFT_DOWN_MASK)) {
+          switch (e.getKeyCode()) {
+            case KeyEvent.VK_PLUS:
+            case KeyEvent.VK_EQUALS:
+              base.handleFontSizeChange(1);
+              break;
+            case KeyEvent.VK_MINUS:
+              if (!e.isShiftDown()) {
+                base.handleFontSizeChange(-1);
+              }
+              break;
+          }
+        }
+      }
+    });
   }
 
   public void applyPreferences() {
+
+    // Update the console text pane font from the preferences.
     Font consoleFont = Theme.getFont("console.font");
     Font editorFont = PreferencesData.getFont("editor.font");
     Font actualFont = new Font(consoleFont.getName(), consoleFont.getStyle(), scale(editorFont.getSize()));
 
+    AttributeSet stdOutStyleOld = stdOutStyle.copyAttributes();
+    AttributeSet stdErrStyleOld = stdErrStyle.copyAttributes();
     StyleConstants.setFontSize(stdOutStyle, actualFont.getSize());
     StyleConstants.setFontSize(stdErrStyle, actualFont.getSize());
 
-    out.setAttibutes(stdOutStyle);
-    err.setAttibutes(stdErrStyle);
+    // Re-insert console text with the new preferences if there were changes.
+    // This assumes that the document has single-child paragraphs (default).
+    if (!stdOutStyle.isEqual(stdOutStyleOld) || !stdErrStyle.isEqual(stdOutStyleOld)) {
+      out.setAttibutes(stdOutStyle);
+      err.setAttibutes(stdErrStyle);
+
+      int start;
+      for (int end = document.getLength() - 1; end >= 0; end = start - 1) {
+        Element elem = document.getParagraphElement(end);
+        start = elem.getStartOffset();
+        AttributeSet attrs = elem.getElement(0).getAttributes();
+        AttributeSet newAttrs;
+        if (attrs.isEqual(stdErrStyleOld)) {
+          newAttrs = stdErrStyle;
+        } else if (attrs.isEqual(stdOutStyleOld)) {
+          newAttrs = stdOutStyle;
+        } else {
+          continue;
+        }
+        try {
+          String text = document.getText(start, end - start);
+          document.remove(start, end - start);
+          document.insertString(start, text, newAttrs);
+        } catch (BadLocationException e) {
+          // Should only happen when text is async removed (through clear()).
+          // Accept this case, but throw an error when text could mess up.
+          if (document.getLength() != 0) {
+            throw new Error(e);
+          }
+        }
+      }
+    }
```

Comment:
May you explain why you do this loop backwards? There is no other way to do it all at once (I guess no because the text may have multiple attributes like colors etc.)?

Reply:
I've tried looping forward, but since the text is taken out and re-inserted, getting the next offset will bug out. I am uncertain why this happens, but looping from back to front ensures that re-inserted parts will never be visited twice or affect next parts in the loop.

Reply:
And indeed, the text has to be handled per document element due to it containing info and error text, which has a different color that should be maintained.